x-pack/plugins/synthetics/public/legacy_uptime/components/monitor_management/manage_locations/manage_locations_flyout.tsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect, useState } from 'react';
+import {
+  EuiFlyout,
+  EuiButtonEmpty,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiSpacer,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiCallOut,
+} from '@elastic/eui';
+import { useDispatch, useSelector } from 'react-redux';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { ClientPluginsStart } from '../../../../plugin';
+import { EmptyLocations } from './empty_locations';
+import { getServiceLocations } from '../../../state/actions';
+import { LocationForm } from './location_form';
+import { PrivateLocationsList } from './locations_list';
+import { useLocationsAPI } from './hooks/use_locations_api';
+import {
+  getAgentPoliciesAction,
+  selectManageFlyoutOpen,
+  setManageFlyoutOpen,
+} from '../../../state/private_locations';
+
+export const ManageLocationsFlyout = () => {
+  const [isAddingNew, setIsAddingNew] = useState(false);
+
+  const dispatch = useDispatch();
+
+  const setIsOpen = (val: boolean) => dispatch(setManageFlyoutOpen(val));
+
+  const isOpen = useSelector(selectManageFlyoutOpen);
+
+  const { onSubmit, loading, privateLocations, onDelete } = useLocationsAPI({ isOpen });
+
+  const { fleet } = useKibana<ClientPluginsStart>().services;
+
+  const hasFleetPermissions = Boolean(fleet?.authz.fleet.readAgentPolicies);
```

Comment:
Do you think we should also require integrations write permissions, since adding a location is pointless without it. Wondering your thoughts here.

Reply:
i think in this operation we only read permissions so it makes more sense to limit it to that.